File: packages/dotnet/CatalystSharp/Models/NotificationGroup.cs
using System.Text.Json.Serialization;

namespace CatalystSharp.Models;

public record NotificationAdditionalContext(
    [property: JsonPropertyName("type")] string Type,
    [property: JsonPropertyName("url")] string Url,
    [property: JsonPropertyName("format")] string Format
);

public record NotificationGroup(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("body")] string Body,
    [property: JsonPropertyName("occurredBy")]
    EgeriaUser OccurredBy,
    [property: JsonPropertyName("isRead")] bool IsRead,
    [property: JsonPropertyName("createdAt")] string CreatedAt,
    [property: JsonPropertyName("additionalContexts")]
    NotificationAdditionalContext? AdditionalContexts = null
);

File: packages/swift/Sources/CatalystSwift/Models/NotificationGroup.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

public struct NotificationAdditionalContext: Decodable, Sendable, Equatable, Hashable {
  public let type: String
  public let url: String
  public let format: String
}

public struct NotificationGroup: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let body: String
  public let occurredBy: EgeriaUser
  public let isRead: Bool
  public let createdAt: String
  public let additionalContexts: NotificationAdditionalContext?
}
